### DISCUSSION TITLE
Difference between plugins/*.server.ts and server/plugins/*.ts

### DISCUSSION THREAD

**Discussion by mogi86** [nuxt/nuxt#19288] [Questions] [answered]
I know Nuxt can define such files.

- `plugins/*.server.ts`
- `server/plugins/*.ts`

However, I don't know difference between `plugins/*.server.ts` and `server/plugins/*.ts` What is difference? And are there any best practices for using them differently?

**Comment by dsfcoll** [comment, marked as answer]
`server/plugins/*.ts` is a nitro plugins, which called earlier only once on nitro startup. They are called only on server-side. For example can be used if you have to add dynamically routes to the server. Or to subscribe to `nitroApp` hooks.
Also because it is called early, it doesn't have access to nuxt composables and therefore to store too.

`plugins/*.server.ts` is a nuxt plugin which called later and on every user request. They have access to composables, `vueApp`, `nitroApp`, `store`.

**Reply by mogi86** [reply]
Thank you. I understand 😄

**Comment by jirehnimes** [comment]
@VladStepanov could you possibly know how to use the plugins created under `server/plugins/*.ts`? The documentation only shows `console.log` both in Nuxt and Nitro. Let say the created plugin will be used in `server/api/**/*.ts`?

Related discussions:
https://github.com/nuxt/nuxt/discussions/20114
https://github.com/unjs/nitro/discussions/1221
